### PR TITLE
WELD-2353 Event.fireAsync() - make it possible to specify that observer methods should be notified in parallel

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

File: environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/event/options/parallel/ParallelNotificationModeTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+/**
+ *
+ * @author Martin Kouba
+ */
+@RunWith(Arquillian.class)
+public class ParallelNotificationModeTest {
+
+    @Deployment
+    public static Archive<?> createTestArchive() {
+        return ClassPath.builder().add(ShrinkWrap.create(BeanArchive.class, Utils.getDeploymentNameAsHash(WeldInstanceTest.class))
```
Comment (tremes) [inline environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/event/options/parallel/ParallelNotificationModeTest.java]: Just noticed the minor here - referencing another test in `Utils.getDeploymentNameAsHash`

File: environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/event/options/parallel/ParallelNotificationModeTest.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            assertEquals(2, threadNames.size());
+        }
+    }
+
+    @Test
+    public void testExceptionPropagated() throws InterruptedException {
+        try (WeldContainer container = createWeld()) {
+            BlockingQueue<Throwable> synchronizer = new LinkedBlockingQueue<>();
+            container.event().select(Message.class).fireAsync(() -> {
+                throw new IllegalStateException(ParallelNotificationModeTest.class.getName());
```
Comment (tremes) [inline environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/event/options/parallel/ParallelNotificationModeTest.java]: What about a test which would throw some exception directly in the observer method?

**Comment by tremes** [comment]
Otherwise it looks reasonable.

File: environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/event/options/parallel/ParallelNotificationModeTest.java (hunk 2 above)
Comment (tremes) [inline environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/event/options/parallel/ParallelNotificationModeTest.java]: Ahh I realized it's basically it...so ignore please and viva the lambdas.

File: environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/event/options/parallel/ParallelNotificationModeTest.java (hunk 1 above)
Comment (mkouba) [inline environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/event/options/parallel/ParallelNotificationModeTest.java]: Good catch!

**Comment by mkouba** [comment]
@antoinesd Your opinion on this PR?

**Pull request merged**